lib/m2ts/m2ts.js

```diff
@@ -406,6 +407,40 @@ ElementaryStream = function() {
         stream.data.length = 0;
       }
 
+      if (!segmentHadPmt) {
+        var
+          pmt = {
+            type: 'metadata',
+            tracks: []
+          };
+        // translate audio and video streams to tracks
+        if (programMapTable.video !== null) {
```

Comment:
looks like we need to null check `programMapTable` itself here and below too, likely.

Reply:
good call, I also realized that flushStreams isn't only called on `flush` which gave us a lot more `trackinfo` events than we wanted. I moved this code to `this.flush` and the unit tests in vhs pass 💯